Analysis/Tasks/PWGHF/taskX.cxx

```diff
@@ -0,0 +1,87 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file taskX.cxx
+/// \brief X(3872) analysis task
+///
+/// \author Gian Michele Innocenti <gian.michele.innocenti@cern.ch>, CERN
+/// \author Rik Spijkers <r.spijkers@students.uu.nl>, Utrecht University
+
+#include "Framework/AnalysisTask.h"
+#include "Framework/HistogramRegistry.h"
+#include "AnalysisDataModel/HFSecondaryVertex.h"
+#include "AnalysisDataModel/HFCandidateSelectionTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::aod::hf_cand_prong2;
+using namespace o2::framework::expressions;
+
+void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
+{
+  ConfigParamSpec optionDoMC{"doMC", VariantType::Bool, false, {"Fill MC histograms."}};
+  workflowOptions.push_back(optionDoMC);
+}
+
+#include "Framework/runDataProcessing.h"
+
+/// X(3872) analysis task
+struct TaskX {
+  HistogramRegistry registry{
+    "registry",
+    {{"hMassJpsi", "2-prong candidates;inv. mass (e+ e-) (GeV/#it{c}^{2});entries", {HistType::kTH1F, {{500, 0., 5.}}}},
+     {"hPtCand", "X candidates;candidate #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}}}};
+
+  Configurable<int> selectionFlagJpsi{"selectionFlagJpsi", 1, "Selection Flag for Jpsi"};
+  Configurable<double> cutYCandMax{"cutYCandMax", -1., "max. cand. rapidity"};
+
+  Filter filterSelectCandidates = (aod::hf_selcandidate_jpsi::isSelJpsiToEE >= selectionFlagJpsi);
+
+  /// aod::BigTracks is not soa::Filtered, should be added when filters are added
+  void process(aod::Collision const&, aod::BigTracks const& tracks, soa::Filtered<soa::Join<aod::HfCandProng2, aod::HFSelJpsiToEECandidate>> const& candidates)
+  {
+    for (auto& candidate : candidates) {
+      if (!(candidate.hfflag() & 1 << JpsiToEE)) {
+        continue;
+      }
+      if (cutYCandMax >= 0. && std::abs(YJpsi(candidate)) > cutYCandMax) {
+        continue;
+      }
+      registry.fill(HIST("hMassJpsi"), InvMassJpsiToEE(candidate));
+
+      int index0jpsi = candidate.index0Id();
+      int index1jpsi = candidate.index1Id();
+      for (auto& trackPos : tracks) {
+        if (trackPos.signed1Pt() < 0) {
+          continue;
+        }
+        if (trackPos.globalIndex() == index0jpsi) {
+          continue;
+        }
+        for (auto& trackNeg : tracks) {
+          if (trackNeg.signed1Pt() > 0) {
+            continue;
+          }
+          if (trackNeg.globalIndex() == index1jpsi) {
+            continue;
+          }
```

Comment:
You could reduce the number of iterations from N^2 down to N(N-1)/2 by iterating over all tracks in the first loop and iterating over `track2 = track1 + 1` in the second loop. You just need to check once that `track1.signed1Pt() != track2.signed2Pt().`